client.go

```diff
@@ -55,32 +58,91 @@ type LenReader interface {
 type Request struct {
 	// body is a seekable reader over the request body payload. This is
 	// used to rewind the request data in between retries.
-	body io.ReadSeeker
+	body func() io.Reader
```

Comment:
This would probably be better as `func() (io.Reader, error)`, so if you are doing something like opening a file you have an out in error cases. You could pass `func() (io.Reader, error) { return os.Open("foo.txt") }` then, for example.

Reply:
Fair enough, I'll change that.